### PR TITLE
Add person, student, teacher

### PR DESCRIPTION
**Pull request by MattGomb**
Dear Reviewer,

Thank you for taking the time to review my project.

In this PR I created the start of the project [OOP school library](https://github.com/MattGomb/rubypractice/blob/master/animal.rb), according to the requirements:
- I created separate files: person, student, teacher
- In the Person class I created the constructor and the instance variables.
- I created the getters and setters, but rubocop insisted to change `:name` and `:age` to `attr_accessor` 
- I created the private and public methods
- I linked `Student` and `Teacher` to `Person` and gave both of the modified constructors, and each of them got their own methods.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
```
Comment (brytebee) [inline person.rb]: - [x] **Optional:** In future projects, kindly please your attr at the top of the class before the initialize method. Although the functionalities are the same, it's just a Ruby way/convention,

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def can_use_services?
+    case of_age? || @parent_permission
+    when true
+      true
+    when false
+      false
+    end
+  end
```
Comment (brytebee) [inline person.rb:21]: - [x] **Required:** kindly move this method above the private keyword as this is required to be public. All methods beneath the private keyword as in your case are private.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @MattGomb👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- No linter errors.✔️
- Used the required Git flow.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (MattGomb) [reply person.rb:21]: Thanks, done!

File: person.rb (hunk 1 above)
Comment (MattGomb) [reply person.rb]: Done!

**Pull request merged**